GitCommands/RevisionReader.cs

```diff
@@ -301,18 +314,35 @@ private bool TryParseRevision(in ReadOnlySpan<byte> buffer, [NotNullWhen(returnV
             #region Object ID, Tree ID, Parent IDs
 
             // The first 40 bytes are the revision ID and the tree ID back to back
-            if (!ObjectId.TryParse(buffer.Slice(0, ObjectId.Sha1CharCount), out ObjectId? objectId) ||
-                !ObjectId.TryParse(buffer.Slice(ObjectId.Sha1CharCount, ObjectId.Sha1CharCount), out ObjectId? treeId))
+            ReadOnlyMemory<byte> commitHash = buffer.Slice(0, ObjectId.Sha1CharCount);
+            ReadOnlySpan<byte> commitHashSpan = commitHash.Span;
+            ObjectId? objectId;
+            if (_cache.objectId != null && commitHashSpan.SequenceEqual(_cache.buffer.Span))
+            {
+                objectId = _cache.objectId;
```

Comment:
ObjectId.TryParse() should be faster with .net9, there could be other improvements too (how to enable vector processing?)
But also nice improvement.